src/Compilers/CSharp/Portable/CSharpCompilationOptions.cs

```diff
@@ -331,165 +257,87 @@ public new CSharpCompilationOptions WithPublicSign(bool publicSign)
-
-            return new CSharpCompilationOptions(this) { MetadataImportOptions_internal_protected_set = value };
+            return (value == this.MetadataImportOptions) ? this : new CSharpCompilationOptions(this) { MetadataImportOptions_internal_protected_set = value };
         }
 
         internal CSharpCompilationOptions WithReferencesSupersedeLowerVersions(bool value)
         {
-            if (value == this.ReferencesSupersedeLowerVersions)
-            {
-                return this;
-            }
-
-            return new CSharpCompilationOptions(this) { ReferencesSupersedeLowerVersions_internal_protected_set = value };
+            return (value == this.ReferencesSupersedeLowerVersions) ? this : new CSharpCompilationOptions(this) { ReferencesSupersedeLowerVersions_internal_protected_set = value };
         }
 
         public new CSharpCompilationOptions WithXmlReferenceResolver(XmlReferenceResolver resolver)
         {
-            if (ReferenceEquals(resolver, this.XmlReferenceResolver))
-            {
-                return this;
-            }
-
-            return new CSharpCompilationOptions(this) { XmlReferenceResolver = resolver };
+            return (ReferenceEquals(resolver, this.XmlReferenceResolver)) ? this : new CSharpCompilationOptions(this) { XmlReferenceResolver = resolver };
         }
 
         public new CSharpCompilationOptions WithSourceReferenceResolver(SourceReferenceResolver resolver)
         {
-            if (ReferenceEquals(resolver, this.SourceReferenceResolver))
-            {
-                return this;
-            }
-
-            return new CSharpCompilationOptions(this) { SourceReferenceResolver = resolver };
+            return (ReferenceEquals(resolver, this.SourceReferenceResolver)) ? this : new CSharpCompilationOptions(this) { SourceReferenceResolver = resolver };
         }
 
         public new CSharpCompilationOptions WithMetadataReferenceResolver(MetadataReferenceResolver resolver)
         {
-            if (ReferenceEquals(resolver, this.MetadataReferenceResolver))
-            {
-                return this;
-            }
-
-            return new CSharpCompilationOptions(this) { MetadataReferenceResolver = resolver };
+            return (ReferenceEquals(resolver, this.MetadataReferenceResolver)) ? this : new CSharpCompilationOptions(this) { MetadataReferenceResolver = resolver };
         }
 
         public new CSharpCompilationOptions WithAssemblyIdentityComparer(AssemblyIdentityComparer comparer)
         {
             comparer = comparer ?? AssemblyIdentityComparer.Default;
 
-            if (ReferenceEquals(comparer, this.AssemblyIdentityComparer))
-            {
-                return this;
-            }
-
-            return new CSharpCompilationOptions(this) { AssemblyIdentityComparer = comparer };
+            return (ReferenceEquals(comparer, this.AssemblyIdentityComparer)) ? this : new CSharpCompilationOptions(this) { AssemblyIdentityComparer = comparer };
```

Comment:
Un-necessary parens, especially when no `==`.
Consider breaking long lines (I generally aim for 120-130 chars, but I don't think we have a hard rule).